File: gems/decomplex/examples/cpp/semantic-alias.cpp
bool frame(Node node) { return node.provenance == FRAME; }
bool is_frame(Node node) { return provenance == FRAME; }
bool heap(Node node) { return node.provenance == HEAP; }
int somewhere(Node node) { if (node.provenance == FRAME) { return 1; } return 0; }

File: gems/decomplex/examples/kotlin/semantic-alias.kt
fun frame(node: Node): Boolean { return node.provenance == FRAME }
fun is_frame(node: Node): Boolean { return provenance == FRAME }
fun heap(node: Node): Boolean { return node.provenance == HEAP }
fun somewhere(node: Node): Int { if (node.provenance == FRAME) { return 1 }; return 0 }
